user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -141,15 +141,13 @@ def _process_offline_cluster_args(self) -> None:
 
     def __process_filter_args(self, arg_val: str) -> None:
         selected_filter = QualFilterApp.fromstring(arg_val)
-        if selected_filter is None:
-            selected_filter = QualFilterApp.get_default()
-            available_filters = [filter_enum.value for filter_enum in QualFilterApp]
-            self.logger.warning(
-                'Invalid argument filter_apps=%s.\n\t'
-                'Accepted options are: [%s].\n\t'
-                'Falling-back to default filter: %s',
-                arg_val, Utils.gen_joined_str(' | ', available_filters),
-                QualFilterApp.tostring(selected_filter))
+        available_filters = [filter_enum.value for filter_enum in QualFilterApp]
+        self.logger.warning(
+            'Invalid argument filter_apps=%s.\n\t'
+            'Accepted options are: [%s].\n\t'
+            'Falling-back to default filter: %s',
+            arg_val, Utils.gen_joined_str(' | ', available_filters),
+            QualFilterApp.tostring(selected_filter))
```

Comment:
Is this behavior the same? Previously, this block does not raise exception even if `arg_val` is not `QualFilterApp`, but after this changes, it could.

Reply:
This check seemed redundant to me.  `QualFilterApp` argument is provided by the user, and our argument validator (argprocessor) would validates these inputs as first step. If there are invalid values, we will terminate early.

Reply:
This code was initially there because we had the legacy CLI that was not doing the same initialization as `spark_rapids`.
There pros-cons removing it:

- pros: it is redundant
- cons: Starting the qualification tool without going through the argProcessor will bypass all those checks.

Reply:
Yes, in that case, it will bypass those checks. Now, if caller provides invalid value of `QualFilterApp`, the tool would crash instead of silently continuing with the default value. I think this should be the right behavior. I agree this differs from the original behavior.